### DISCUSSION TITLE
ENV to remove antispam from SMTP milters conf

### DISCUSSION THREAD

**Discussion by 01e9** [Mailu/Mailu#3617] [User support]
I host Mailu on a cheap VPS and I need as low RAM consumption as possible.

In docker-compose I commented some containers, including the `antispam` container.
_Now I have these conainers left: `front`, `admin`, `smtp`, `imap`, `resolver`, `edis`._

Everything works fine except email sending, because SMTP requires antispam.

I changed directly in container [`/conf/main.cf`](https://github.com/Mailu/Mailu/blob/4ca111964b0c8c3402962e53a0102907c63c286a/core/postfix/conf/main.cf#L133C1-L133C14)
```diff
-smtpd_milters = inet:{{ ANTISPAM_ADDRESS }}:11332
+smtpd_milters =
```
restarted the container and email sending works.

Would be nice to have an env variable to disable antispam completely.

**Comment by 01e9** [comment]
Another workaround is to mount the `main.cf`:

1. `docker compose cp smtp:/conf/main.cf smtp.main.cf`
2. Edit and set `smtpd_milters =`
3. Edit `docker-compose.yaml`

    ```diff
    smtp:
        volumes:
    +   - "./smtp.main.cf:/conf/main.cf:ro"
    ```

**Comment by Diman0** [comment]
This is not supported. Rspamd (antispam) is required for core functionality such as providing the headers for SRS, dkim signing, arc siging, dmarc signing. These are core technologies for sending secure email. 

If you enable swap and don't activate any of the optional components, then it is possible to run Mailu on VPS with a lower amount of RAM. 

I'm moving this to discussion since this is not a valid bug report

**Reply by 01e9** [reply]
After upgrading to latest version (from 2.0) the email stopped working for a few days (randomly).

I noticed higher memory usage and decided to disable optional stuff.

I also added in cron weekly containers restart (to reset memory, I don't trust Python garbage collector).

I will enable back antispam.
Thanks
